app/models/king.rb

```diff
@@ -13,6 +17,35 @@ def valid_move?(x,y)
     return false
   end
 
+
+  def can_castle?(x,y)
+    updated_at == created_at && \
```

Comment:
This method is a little complex - you might consider breaking it into smaller named methods like say `never_moved?` and `castle_requested?`, something like (untested pseudocode):

```ruby
class Piece < ActiveRecord::Base
  def never_moved?
    created_at == updated_at
  end
end

class King < Piece
  def castle_requested?(x, y)
    never_moved? && (x == 6 || x == 2)
  end
end
```

Then `can_castle?` can be a combination of smaller named methods and be easier to understand / test.

Also, as noted below on `castle_rook`, you'll want to take into account the case where the first rook you find isn't the castling rook, like if the castle direction was like so:

```
        ---->
R _ _ _ K _ _ R
```

You'll want to make sure to check / move the correct rook based on the castle direction, as a `.first` might pick up the rook on the left when the one on the right is supposed to move.